### PR TITLE
Add Pause Menu and finish individual Room drawing.

### PR DESCRIPTION
**Pull request by Bailey-Ardrey**
Add a PauseMenu class that allows the user to teleport easily to another room by clicking on the room with the mouse. I added a new Zelda_DungeonMap.png that displays the map in the pause menu.

I also finished the room loading for rooms 8, 13, 14, and 18, and made slight changes to the Final Boss enemy movement. I added a new Zelda_Eagle-Map.png to get the text that appears in Room 8.

### CODE REVIEW

**Comment by Bailey-Ardrey** [comment]
One bug is still present when the game is paused - the Goriya boomerang no longer moves. It used to... I am not sure if there were any changes to the boomerang that would have caused this bug. If the boomerang is in the list of enemies (and it is) then the position should change within the PauseMenu ChangePosition() method... but it does not. I will add it to the bug list on the project board.

**Comment by Bailey-Ardrey** [comment]
The above 4 commits do accomplish two things. Firstly, they set a starting position for Link each time a room is cycled through or teleported to from the pause menu. Secondly, they set up a room boundary for the entities in room 13.

File: Sprintfinity3902/Interfaces/IDungeon.cs [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-using Sprintfinity3902.Sprites;
-using System;
-using System.Collections.Generic;
-using System.Text;
 
 namespace Sprintfinity3902.Interfaces
 {
     public interface IDungeon
     {
+        public IRoom CurrentRoom { get; set; }
```
Comment (kylegessel) [inline Sprintfinity3902/Interfaces/IDungeon.cs:8]: I think the method below can help to keep this variable private to the Dungeon entities.

**Review by kylegessel — APPROVED** [review]
I'm going to test, then merge. Really exciting changes.

Edit: the pause menu is really fun to mess with.

**Pull request merged**